packages/relay-compiler/generic/core/RelayWatchmanClient.js

```diff
@@ -17,6 +17,20 @@ const watchman = require('fb-watchman');
 class RelayWatchmanClient {
   _client: any;
 
+  static isAvailable(): Promise<boolean> {
+    return new Promise((resolve) => {
+      const client = new RelayWatchmanClient();
+      client.on('error', () => {
+        resolve(false);
+      });
+      client.hasCapability('relative_root')
+        .then(() => {
+          client.end();
```

Comment:
Should we really be creating a client, and then ending it like this? Only to create it again in this success case?

Perhaps a better API would be something like `createIfAvailable()` which promises to resolve to an instance of RelayWatchmanClient if "available" (has a relative_root) or otherwise returns null, which could then be detected in the code above